### PR TITLE
Picard CollectGcBiasMetrics stand alone script

### PR DESCRIPTION
**Pull request by hopedisastro**
Calculates GC metrics for a CRAM file using Picard CollectGcBiasMetrics.
Successful batch - https://batch.hail.populationgenomics.org.au/batches/439455/jobs/1

### CODE REVIEW

**Comment by hopedisastro** [comment]
Unfamiliar with picard, but would it be possible to drop down the resource requirements to make the run cheaper? I just requested the same amount of resources used by jobs in `picard.py`.

**Comment by michael-harper** [comment]
Just did a bit of scrounging around and saw SnakeMake documentation requests 1Gb (https://snakemake-wrappers.readthedocs.io/en/0.75.0/wrappers/picard/collectgcbiasmetrics.html) though it's not clear what data that would be applicable to but to me it suggests we could get away with a `STANDARD` machine?

File: str/helper/compute_gc_metrics.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    for id in list(cram_dict.keys()):
+
+        input_cram = b.read_input(cram_dict[id])
+
+        j = b.new_job(f'CollectGcBiasMetrics: {id}')
+
+        j.image(image_path('picard'))
+        resource = STANDARD.request_resources(ncpu=1)
+        resource.attach_disk_storage_gb = 50
+        resource.set_to_job(j)
```
Comment (MattWellie) [inline str/helper/compute_gc_metrics.py:52]: You can make this storage reservation super tight:

```
from cpg_utils import to_path

cram_file = to_path(cram_dict[id])
cram_size = cram_file.stat().st_size
job.storage(cram_size * 1.5)  # arbitrary amount of storage as a buffer

File: str/helper/compute_gc_metrics.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        """
+        j.command(command(cmd, monitor_space=True))
+
+        b.write_output(j.output_files, output_path(f'{id}', 'analysis'))
+
+    b.run(wait=False)
+
+
+if __name__ == '__main__':
+    main()  # pylint: disable=no-value-for-parameter,missing-function-docstring
```
Comment (MattWellie) [inline str/helper/compute_gc_metrics.py:80]:
```suggestion
    main()  # pylint: disable=no-value-for-parameter
```

Always put a function docstring!

**Comment by hopedisastro** [comment]
yay, using 1 cpu and standard mem worked: https://batch.hail.populationgenomics.org.au/batches/439618/jobs/1, cost dropped by 1/3

File: str/helper/compute_gc_metrics.py (hunk 1 above)
Comment (hopedisastro) [reply str/helper/compute_gc_metrics.py]: good idea, thanks!

**Comment by hopedisastro** [comment]
dynamic storage sizing is working nicely - https://batch.hail.populationgenomics.org.au/batches/440034/jobs/1

File: str/helper/compute_gc_metrics.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        cmd = f"""
+
+
+        picard CollectGcBiasMetrics -Xms{resource.get_java_mem_mb()}M \\
+        I={input_cram} O={j.output_files['gc_bias_sample.txt']} CHART={j.output_files['gc_bias_chart.pdf']} \\
+        S={j.output_files['gc_bias_summary.txt']} \\
+        R={reference.base}
+        echo "CollectGcBiasMetrics finished successfully"
+
```
Comment (MattWellie) [inline str/helper/compute_gc_metrics.py:74]:
```suggestion
        picard CollectGcBiasMetrics -Xms{resource.get_java_mem_mb()}M \\
        -I {input_cram} -O {j.output_files['gc_bias_sample.txt']} -CHART {j.output_files['gc_bias_chart.pdf']} \\
        -S {j.output_files['gc_bias_summary.txt']} \\
        -R {reference.base}
        echo "CollectGcBiasMetrics finished successfully"
```

The logging in the [job](https://batch.hail.populationgenomics.org.au/batches/440034/jobs/1) indicates that you should be using the update command line syntax
https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)

**Review by MattWellie — CHANGES_REQUESTED** [review]
just one quick change to the syntax

**Review by MattWellie — APPROVED** [review]
LGTM!

**Pull request merged**